axiom/client.go

```diff
@@ -250,52 +248,12 @@ func (c *Client) NewRequest(ctx context.Context, method, path string, body any)
 // JSON decoded or directly written to v, depending on v being an [io.Writer] or
 // not.
 func (c *Client) Do(req *http.Request, v any) (*Response, error) {
-	var (
-		resp *Response
-		err  error
-	)
-	if req.GetBody != nil && !c.noRetry {
-		bck := backoff.NewExponentialBackOff()
-		bck.InitialInterval = time.Millisecond * 200
-		bck.MaxElapsedTime = time.Second * 10
-		bck.Multiplier = 2.0
-
-		err = backoff.Retry(func() error {
-			var httpResp *http.Response
-			//nolint:bodyclose // The response body is closed later down below.
-			httpResp, err = c.httpClient.Do(req)
-			switch {
-			case errors.Is(err, context.Canceled):
-				return backoff.Permanent(err)
-			case err != nil:
-				return err
-			}
-			resp = newResponse(httpResp)
-
-			// We should only retry in the case the status code is >= 500,
-			// anything below isn't worth retrying.
-			if code := resp.StatusCode; code >= 500 {
-				_, _ = io.Copy(io.Discard, resp.Body)
-				_ = resp.Body.Close()
-
-				// Reset the requests body, so it can be re-read.
-				if req.Body, err = req.GetBody(); err != nil {
-					return backoff.Permanent(err)
-				}
-
-				return fmt.Errorf("got status code %d", code)
-			}
-
-			return nil
-		}, bck)
-	} else {
-		var httpResp *http.Response
-		//nolint:bodyclose // The response body is closed later down below.
-		if httpResp, err = c.httpClient.Do(req); err != nil {
-			return nil, err
-		}
-		resp = newResponse(httpResp)
+	//nolint:bodyclose // The response body is closed later down below.
+	httpResp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
 	}
+	resp := newResponse(httpResp)
 
 	defer func() {
 		if resp != nil {
```

Comment:
Question: when wouldn't the resp be nil here? This defer is instantiated after the check for err, and if err == nil, isn't there by definition a resp? As far as I can see `newResponse(httpResp)` doesn't affect resp being nil.

Reply:
Oh I think because in previous code, it came before the err check?

Reply:
I was just wondering whether it was that that was impacting the linting. But i guess it's to do with the wrapping inside of another struct?

Reply:
Ah, thanks! Yeah, that's definitely code not needed anymore, now :) Cleaned it up and also get rid of the annoying linter comment.